lib/export/jdl_exporter.js

```diff
@@ -18,11 +18,155 @@
  */
 
 const fs = require('fs');
+const path = require('path');
+const applicationOptions = require('../core/jhipster/application_options');
+const unaryOptions = require('../core/jhipster/unary_options');
+const JSONFileReader = require('../reader/json_file_reader');
+const JSONReader = require('../reader/json_reader');
+const JSONToJdlApplicationConverter = require('../converters/json_to_jdl_application_converter');
+const FileUtils = require('../utils/file_utils');
+const JDLObject = require('../core/jdl_object');
 
 module.exports = {
-  exportToJDL
+  exportToJDL,
+  loadJhipsterConfigAndEntities
 };
 
+const excludeUnaryOptions = ['skipUserManagement'];
+
+/**
+ * Indicate if an unary option should be written in the jdl file
+ * @param key
+ */
+function unaryOptionsExists(key) {
+  return !excludeUnaryOptions.includes(key) && unaryOptions.exists(key);
+}
+
+const excludeOptions = ['jhipsterVersion', 'packageFolder', 'otherModules'];
+
+/**
+ * Indicate if an application option should be written in the jdl file
+ * @param key option name
+ */
+function applicationOptionsExists(key) {
+  return !excludeOptions.includes(key) && applicationOptions.exists(key);
+}
+
+/**
+ * Remove unwanted application options for the jdl file
+ * @param application the application
+ */
+function filterApplicationOptions(application) {
+  if (application.config === undefined) {
+    throw new Error(`'${application.toString()}' must contain config field.`);
+  }
+  const config = application.config;
+  Object.keys(config).forEach(key => {
+    if (!applicationOptionsExists(key)) {
+      delete config[key];
+    } else if ((key === 'entitySuffix' || key === 'dtoSuffix') && config[key] === '') {
+      config[key] = false;
+    }
+  });
+  return application;
+}
+
+/**
+ * load the application configuration and the associated entities from a Jhipster project
+ * contained in a folder
+ * @param dir path to the directory where it should contains the .yo-rc.json
+ */
+function loadApplicationAndEntities(dir) {
+  if (!dir) {
+    dir = '.';
+  }
+
+  if (!FileUtils.doesFileExist(path.join(dir, '.yo-rc.json'))) {
+    throw new Error('The directory does not contain a .yo-rc.json');
+  }
+  const jsonapplication = JSONFileReader.readEntityJSON(path.join(dir, '.yo-rc.json'));
+  let jdlapplication = JSONToJdlApplicationConverter.convertApplicationsToJDL({
+    applications: [jsonapplication]
+  }).applications;
+  jdlapplication = jdlapplication[Object.keys(jdlapplication)[0]];
+  let jdlentities = new JDLObject();
+
+  if (FileUtils.doesDirectoryExist(path.join(dir, '.jhipster'))) {
+    jdlentities = JSONReader.parseFromDir(dir);
+  }
+
+  addEntitiesToJdlApplication(jdlentities, jdlapplication);
+  filterApplicationOptions(jdlapplication);
+  return {
+    JDLApplication: jdlapplication,
+    JDLEntities: jdlentities
+  };
+}
+
+/**
+ * Add the entity names used by the application
+ * @param entitiesjdl entities
```

Comment:
Can you be a little bit more verbose here? :)

Reply:
sure!